File: GuideObjC/GuideObjC/ViewControllers/LoginController/LoginController.m
//
//  LoginController.m
//  GuideObjC
//
//  Created by LuzanovRoman on 09.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "LoginController.h"
#import "ProgressController.h"

@interface LoginController ()
@property (weak, nonatomic) IBOutlet UILabel *errorLabel;
@property (weak, nonatomic) IBOutlet UITextField *emailTextField;
@property (weak, nonatomic) IBOutlet UITextField *passwordTextField;
@end

@implementation LoginController


- (IBAction)login:(id)sender {
    _errorLabel.text = nil;
    
    NSString * email = _emailTextField.text;
    NSString * password = _passwordTextField.text;
    
    if (email.length == 0 || password.length == 0) {
        _errorLabel.text = @"Enter email and password";
        return;
    }
    [self.view endEditing:true];
    
    [ProgressController show];
    [self.graph loginWithEmail:email password:password completion:^(id object, NSError * error) {
        
        if (error) {
            self.errorLabel.text = error.localizedFailureReason;
            [ProgressController hide];
        }
        else {
            [self getUserObject];
        }
    }];
}

- (void)getUserObject {
    [self.graph userObjectWithCompletion:^(NSObject * object, NSError * error) {
        [ProgressController hide];
        
        if (error) {
            self.errorLabel.text = [NSString stringWithFormat:@"Can't get user. %@", error.localizedDescription];
        }
        else {
            [self dismissViewControllerAnimated:true completion:nil];
        }
    }];
}

// MARK:- UITextFieldDelegate
- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [self.view endEditing:true];
    return true;
}
@end

File: GuideSwift/GuideSwift/ViewControllers/LoginController/LoginController.swift
//
//  LoginController.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 09.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

class LoginController: UIViewController, UITextFieldDelegate {
    
    @IBOutlet weak var errorLabel: UILabel!
    @IBOutlet weak var emailTextField: UITextField!
    @IBOutlet weak var passwordTextField: UITextField!
    
    @IBAction func login(_ sender: AnyObject) {
        errorLabel.text = nil
        view.endEditing(true)
        
        guard let email = emailTextField.text, let password = passwordTextField.text, email.isEmpty == false, password.isEmpty == false else {
            errorLabel.text = "Enter email and password"
            return
        }
        ProgressController.show()
        Graph.login(withEmail: email, password: password) { (object, error) in
            
            if let err = error {
                self.errorLabel.text = err.localizedFailureReason
                ProgressController.hide()
            }
            else {
                self.getUserObject()
            }
        }
    }
    
    func getUserObject() {
        Graph.userObject { (object, error) in
            ProgressController.hide()
            
            if let err = error {
                self.errorLabel.text = "Can't get user. \(err.localizedDescription)"
            }
            else {
                self.dismiss(animated: true, completion: nil)
            }
        }
    }
    
    // MARK: UITextFieldDelegate
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        view.endEditing(true)
        return true
    }
}
